### ISSUE TITLE
Generalize chunks/ready_chunks/try_chunks/etc to collect into T: Extend, not just Vec

### ISSUE THREAD

**Issue by khuey** [rust-lang/futures-rs#2492]
In particular I would find it really useful to be able to ready_chunks a Stream of Futures into FuturesOrdereds and spawn them on a threadpool to parallelize my computation.

This would be a breaking change of course.

**Comment by taiki-e** [comment]
Seems reasonable to me.

**Comment by khuey** [comment]
This is actually a bit harder than I thought. Ideally there would be traits for `with_capacity` and `len` too so that we can reserve space in the collection and not have to track the collection's size again ourselves in `Chunks`.

**Comment by ibraheemdev** [comment]
There is `extend_reserve` on nightly.

**Comment by khuey** [comment]
Yeah people don't seem eager to stabilize it though, and there's nothing at all for `len`.
